HelpSource/Reference/JITLib_reblock.schelp

```diff
@@ -0,0 +1,213 @@
+s.checkUpsample(3, "test");
+// also true, can be nil so one can clear a blockSize value
+s.checkBlockSize(nil)
+::
+
+section:: ProxySpace, Ndef, NodeProxy
+
+code::
+// When nothing is set, Ndefs just uses server blockSize
+Ndef(\x, { BlockSize.ir.poll(0); LFSaw.ar(123, 0, 0.3) });
+Ndef(\x).play;
+::
+
+
+subsection:: NodeProxy slot objects
+The lowest level to set blockSize and upsample is at the objects
+within nodeproxy/Ndef objects, the slots for synthesis processes.
+
+code::
+// turn on dumpOSC to check which messages the server gets
+s.dumpOSC;
+
+Ndef(\x).put(0, { BlockSize.ir.poll(0); LFSaw.ar(123, 0, 0.3) }, blockSize: 32);
+
+// dumpOSC shows: the message begins with 66, which is the special \s_newEx key.
+// the args are defname, nodeID, 1, parentGroup, blockSize, upsample, ... synth params.
+// the default value for blockSize and upsample is 0,
+// so .... 32, 0 ... means blockSize is set to 32, upsample uses its default
+[ 66, "temp__0x10315637_1013", 1004, 1, 1000, 32, 0, "out", 4, "fadeTime", 0.02, "i_out", 4 ]
+
+// setting only the upsample value: ... groupID, 0, 2 ...
+Ndef(\x).put(0, { BlockSize.ir.poll(0); LFSaw.ar(123, 0, 0.3) }, upsample: 2);
+[ 66, "temp__0x10315637_1017", 1006, 1, 1000, 0, 2, "out", 4, "fadeTime", 0.02, "i_out", 4 ]
+::
+
+subsection:: ProxySpace
+ProxySpaces can have blockSize upsample values,
+which are passed on to new proxies in that proxyspace.
+
+code::
+Ndef.clear;
+Ndef.dictFor(s) // the proxyspace for all Ndefs on s
+
+Ndef.dictFor(s).blockSize; // nil by default
+Ndef.dictFor(s).upsample; // nil by default
+
+// set and get its blockSize:
+Ndef.dictFor(s).blockSize = 8;
+Ndef.dictFor(s).blockSize; // now 8
+// set and get its upsample factor
+Ndef.dictFor(s).upsample = 8;
+Ndef.dictFor(s).upsample; // now 8
+
+// Newly made Ndefs take these settings from their proxyspace
+Ndef(\y).blockSize;
+Ndef(\y).upsample;
+
+
+// ... and this Ndef then uses its proxyspace's blockSize
+(
+Ndef(\y, {
+	BlockSize.ir.poll(0, "blocksize");
+	SampleRate.ir.poll(0, "samplerate");
+	(SampleRate.ir / s.sampleRate).poll(0, "upsample");
+
+	LFSaw.ar(LFDNoise3.kr([3, 5]).exprange(2000, 20000)) * 0.2
+});
+Ndef(\y).play;
+)
+
+// blockSize and upsample can be cleared again.
+Ndef.dictFor(s).blockSize = nil;  Ndef.dictFor(s).blockSize;
+Ndef.dictFor(s).upsample = nil;   Ndef.dictFor(s).upsample;
+::
+
+subsection:: Ndef, NodeProxy
+
+code::
+// at the mid level, Ndefs can have blockSize and upsample settings
+Ndef(\x).blockSize_(16).upsample_(4);
+Ndef(\x).blockSize;
+Ndef(\x).upsample;
+// which they pass on to the slot objects
+Ndef(\x).put(0, { BlockSize.ir.poll(0); LFSaw.ar(123, 0, 0.3) });
+
+// you can override the Ndef settings at the slot object level
+Ndef(\x).put(0, { BlockSize.ir.poll(0); LFSaw.ar(123, 0, 0.3) }, blockSize: 32);
+
+// blockSize/upsample can be different per slot object:
+Ndef(\x).put(5, { BlockSize.ir.poll(0); LFSaw.ar(123, 0, 0.3) }, upsample: 2);
+::
+
+
+section:: Events and Patterns
```

Comment:
These are not related to jitlib, so maybe the file as a whole could be renamed into a tutorial for all the functionality?

Reply:
Yes. It ended up this way because I expected to do this mainly for PatternProxy, 
but it turned out that it needed to be in all these places in Event, 
and then it worked already. 
So "Using blockSize and upsample in Synths, Events, Patterns and Proxies"?

Reply:
yes, started in that direction. will write this out in more detail, 
would appreciate recommendations and feedback on sketches!

Reply:
Renamed, reorganised guide file, please check if it makes sense as sketched out, then I will write it out more fully.